### ISSUE TITLE
[PvP mechanics] Stealth and unstealthing targets + AoE Spells.

### ISSUE THREAD

**Issue by tedrasiel** [ccshiro/cc-buglist#116]
this report contains 2 Bug Reports.

FIRST:
 BUG: 

Aoe effects like Blizzard, Rain of fire and co. are unstealthing Rogues instantly (when the AOE range hits the rogue). Instantly means, that the rogue can be located at the very end of the spell radius, is sprinting outside of it, but get unstealthed. the rogue did NOT get any dmg so far but still gets unstealthed.

WAY IT SHOULD BE: 

There are projectiles inside these AoE spells dealing dmg (like the single shards inside Blizzard), they have to hit the rogue (flying down first) and have to deal actual dmg to break stealth.

Also: the slow (for blizzard) and dmg effect (for rain of fire ) is applied way too fast. While testing it took about 0,5s to apply them. On retail it took about 1s+. (after using the spell ofc).

SECOND:
BUG:
Rogues getting shielded by a priest while having DoTs on them get unstealthed even tho the DoTs dont deal any dmg (thx to the shield).

WAY IT SHOULD BE: Rogues having Dots on them and getting shielded by a priest can stealth (via vanish or normally via leavin combat) and stealth shouldnt break cause of the DoTs as the shield absorbs all damage. The second the shield runs out or is broken due to dmg, the DoTs do deal actual dmg to the rogue and should break stealth.

**Comment by Iliana-sc** [comment]
I think it's better if you create a separate bug report for the second bug so they can be marked separately.

**Comment by SuddenDeath** [comment]
The shards/projectiles has got nothing to do with the damage, it's just an animation.

Didn't search much but I found some footage of both the mage & the receivers perspective.

Frame 1: *Blizzard castbar appears
Frame 2
Frame 3
Frame 4
Frame 5: Combat log "Noone gains Blizzard"
Frame 6
Frame 7
Frame 8
Frame 9
Frame 10
Frame 11
Frame 12
Frame 13
Frame 14
Frame 15 Combat log "You are affected by Blizzard"
Frame 16
Frame 17
Frame 17

"You are affected by Blizzard"

I'm guessing it's just spell delay + latency, meaning the damage/effect is supposed to be instant.
Now the thing that might have caused this misconception is that when a rogue gets "AoEd" out of stealth, from other players perspective there's a delay before they can actually see the rogue, even though the rogue is actually already out of stealth. This only seem to happen when AoE gets the rogue out of stealth and not when the rogue starts attacking someone for example(might be wrong, could be something related to spell delay or latency)

**Comment by Klimpb** [comment]
The shards/projectiles has got nothing to do with the damage, it's just an animation.

Didn't search much but I found footage of both the mage & the rogues perspective.

Acrono Shadowstep Tricks: (6:58)
https://mega.co.nz/#!HQcy1QJL!1pqzxU1F74H7s7w_cHsmc25xjOvzGYWDG7DyGIsY4nw

Frame 1 *Blizzard castbar appears
Frame 5 *Combat log "Noone gains Blizzard"
Frame 15 *Combat log "You are affected by Blizzard"
Frame 21 *Character animation indicates him being hit by something
Frame 26 *76 damage by blizzard
Frame 35 +Combat,
Frame 36 -Stealth, -Vanish

I'm guessing it's just spell delay + latency, meaning the damage/effect is supposed to be instant(or just delayed by 0,1 second or so), then I guess when someone gets "affected" by blizzard, that itself could have some form of spell delay? Which explains why it takes couple of more frames until it deals damage & breaks stealth. I'm guessing the mage could just stop casting blizzard pretty much right away after having casted it, assuming an enemy is in the area of effect, and the enemy would still be damage and brought out of stealth but with a delay as displayed in the video.

Now the thing that might have caused this misconception is that when a rogue gets "AoEd" out of stealth, from other players perspective there seems to be an even longer delay before they can actually see the rogue, even though the rogue is already out of stealth, this can be seen by looking at the area of where the blizzard is at and where the rogue actually appears out of stealth. Many times it will be 10 yards outside the radius. This only seem to happen when AoE gets the rogue out of stealth and not when the rogue starts attacking someone for example(might be wrong, could be something related to spell delay or latency).

This can be seen in Vurtne 60-66 PvP (8:31)
https://mega.co.nz/#!Pc8TlCJY!DbvZPRq5C8Q-MJP0zPhkqQ7XcA4BxH14Hv2spb-ZNyk

in Dilir 21 (4:44) & (5:08 for adamantite grenade hit on rogue in stealth, takes 19 frames from the moment the rogue was hit until you can actually see him(which also coincidentally(?) matches the time from Frame 15 to 36 (2 frame difference) in the Acrono video)
https://mega.co.nz/#!jUcT2D5A!NNETdbBeRz6_BjVWvgxcxa6vhRGMi2z3LzU4i3jKY8Y

and in Noone 3 (17:45)
https://mega.co.nz/#!XENHCJjI!NHv4_IUx9ikfr2JTb60sz7g1nOvIlFretHVQHGCqvlc

Couldn't find a mage video showing combat log, would be nice.

**Comment by tedrasiel** [comment]
its really hard to find an expample for what i mean. But what im talking about is, that blizzard, like other aoe spells, takes some time until deadling actual dmg. u can see that in PvE too. Theres quite a "large" amount of time passing before these spells deal actual dmg. As i know it, and im pretty shure about it, (still searching for proof, not easy to find this situation) theres a chance for blizzards (for expample) animation to "hit" u and not dealin dmg (if u where located at the very end of it) and leavin the AoE radius instantly , resulting in putting you and in this case the mage into combat without breaking stealth, as dmg hasnt applied jet. There was also the chance (when in the middle of the spell radius with nearly no way to escape, so use cloak to keep in stealth and/or shadowstep out of it instantly to open on the target without leaving stealth in first place. in all cases, as i know it, youre put into combat, but only the actual dmg of blizzard (or spells like that) breaks the stealth. 

to find 100% proof for that we need a video of a rogue beein in that situation.

**Comment by tedrasiel** [comment]
also, when talking about CC (CoreCraft) again, all it needed was the spell, the animation didnt even rlly started, i was sprinting towards the mage, i see the corner animation for a split second slightly behind the middle of the rogues model, i was sprinting. it shows <blizzard> and instantly unstealthed. i didnt took any dmg, and the skilled slow effec of blizzard (as everyone skilled it for some reason - maybee  cause its way more effective that way(if it works instantly)) wasnt upon me (the rogue). still unstealthed.

In this kind of moments, i shoul've putten into combat, but shouldnt lose stealth. 

so in conclusion, theres some delay missing regarding this kind of spells. 

im sorry for my bad english.

**Comment by vexxedd** [comment]
Second bug (de-stealthing with dot and having shield on) is wrong. This is how it SHOULD work. Blizzard implemented that feature only in Wotlk.

**Comment by tedrasiel** [comment]
theres something towards the way blizzard should work:

http://www.warcraftmovies.com/movieview.php?id=92583
1:40 - 1:46 , there u can see the time it takes to unstealth the rogue, this could also be because of the shield, but i dont know how the shield worked now in bc thanks to vexxed. maybe u can find patchnotes for that?

**Comment by Klimpb** [comment]
Regarding my post in this thread, I'm not sure I'm 100% on the right track.. Blizzard & Rain of Fire has got a different damage interval I believe.. I no longer think the damage should be instant, but rather that there is a set time on when it should start the damage intervals after having casted it, that may be different on Blizzard & Rain of Fire.

But I think that "being affected" by it (which happens rather instantaneous) should bring you out of stealth, but with that delay that I was mentioning earlier.

**Comment by Barroth** [comment]
After taking a look at the Noone vs Acrono duels, from Acrono's PoV I found this:
~617ms after the castbar appeared Acrono took dmg from the blizzard(at that point the shard just started falling, they were all still above acrono) and he lost stealth shortly after that.

 Doesn't look like there is a delay except spell delay/latency for blizzard.

Edit: The above is incorrect, I overlooked the old meta gem making the cast take 50% less time.(+20% icy veins)

It should tick after each second(blizzard) and after dealing dmg stealth gets removed. The stealth aura removal is delayed a bit though(spell delay probably) so that rogues often only get visible after they moved some yards away from the aoe.

https://www.youtube.com/watch?v=j1xYH72MkWI shows the ticks happen after each sec + spell delay(1:20)
https://www.youtube.com/watch?v=1Om6qB21EqE shows that the stealth gets removed after the damage hit(2:48)
